runtime/lua/vim/lsp/buf.lua

```diff
@@ -839,6 +840,18 @@ function M.code_action(options)
       params = util.make_range_params(win, client.offset_encoding)
     end
     params.context = context
+
+    -- Don't send markup to clients that don't support it.
+    if not client.server_capabilities.textDocument.diagnostic.markupMessageSupport then
+      --- @param diagnostic lsp.Diagnostic
+      context.diagnostics = vim.iter.map(function(diagnostic)
```

Comment:
Would be good to have an answer to https://github.com/microsoft/language-server-protocol/pull/1905#pullrequestreview-1942818691